src/rgw/rgw_gc.cc

```diff
@@ -235,15 +250,39 @@ class RGWGCIOManager {
     index_io.type = IO::IndexIO;
     index_io.index = index;
 
+    // use lambda to assemble list, so it will only get executed if
+    // we're at the appropirate logging level
```

Comment:
this is nice!

Reply:
Yeah, I wish I'd thought of that before for the busy lock logging. I may go back and fix that one as well.

Reply:
ceph's include/types.h provides an overload of `operator<<` for `vector<T>`, so you should be able to print this vector directly with dout() and avoid the stringstream stuff entirely

Reply:
@cbodley to keep commit #s same for downstream, I'm inclined to let this go through and fix with a subsequent PR. Would that be alright w/ you?